falcon/app.py

```diff
@@ -810,28 +899,18 @@ def handle(req, resp, ex, params):
 
         """
 
-        def wrap_old_handler(old_handler):
-            # NOTE(kgriffs): This branch *is* actually tested by
-            #   test_error_handlers.test_handler_signature_shim_asgi() (as
-            #   verified manually via pdb), but for some reason coverage
-            #   tracking isn't picking it up.
-            if iscoroutinefunction(old_handler):  # pragma: no cover
```

Comment:
Is it safe to remove this code path for coroutine functions?

Reply:
this code was a leftover. the asgi.App overloads this method and never calls super, so unless we want to support async in wsgi, this can never happen

Reply:
Isn't this subject to automatic wrapping in our own tests? Tests pass on your branch, so I guess we never exercise that, even if we do have a machinery for that.

Reply:
I don't follow the test part.

I think this was a leftover added while developing asgi, before it was preferred to overload the method